File: JULY-2025/22-07-2025/p1695.kt
class Solution {
    fun maximumUniqueSubarray(nums: IntArray): Int {
        var left = 0; var right = 0

        var sum = 0
        var res = 0
        val numSet = mutableSetOf<Int>()
        while (right < nums.size) {
            if (numSet.contains(nums[right])) {
                numSet.remove(nums[left])
                sum -= nums[left]
                left++
            } else {
                sum += nums[right]
                res = maxOf(res, sum)
                numSet.add(nums[right])

                right++
            }
        }

        return res
    }
}

File: JULY-2025/22-07-2025/p1695.rs
impl Solution {
    pub fn maximum_unique_subarray(nums: Vec<i32>) -> i32 {
        let mut i = 0;
        let mut set= vec![false; 10000];
        let mut res = 0;
        let mut sum = 0;
        for j in 0..nums.len() {
            while set[nums[j] as usize -1] {
                set[nums[i] as usize -1]=false;
                sum -= nums[i];
                i += 1;
            }
            sum += nums[j];
            set[nums[j] as usize -1]=true;
            res = res.max(sum);
        }
        res
    }
}